src/diffusers/pipelines/hunyuan_image/pipeline_hunyuanimage.py

```diff
@@ -0,0 +1,902 @@
+
+                # Step 1: Collect model inputs needed for the guidance method
+                # The `_guider_input_fields` defines which inputs model needs for conditional/unconditional predictions.
+                # e.g. {"encoder_hidden_states": ("prompt_embeds", "negative_prompt_embeds")}
+                # means both prompt_embeds (conditional) and negative_prompt_embeds (unconditional) as inputs.
+                guider_inputs = {}
+                for _, input_names_tuple in self._guider_input_fields.items():
+                    for input_name in input_names_tuple:
+                        guider_inputs[input_name] = locals()[input_name]
+
+                # Step 2: Update guider's internal state for this denoising step
+                guider.set_state(step=i, num_inference_steps=num_inference_steps, timestep=t)
+
+                # Step 3: Prepare batched model inputs based on the guidance method
+                # The guider splits model inputs into separate batches for conditional/unconditional predictions.
+                # For CFG with _guider_input_fields = {"encoder_hidden_states": ("prompt_embeds", "negative_prompt_embeds")}:
+                #   guider_state = [
+                #       {"encoder_hidden_states": prompt_embeds, "__guidance_identifier__": "pred_cond"},      # conditional batch
+                #       {"encoder_hidden_states": negative_prompt_embeds, "__guidance_identifier__": "pred_uncond"},  # unconditional batch
+                #   ]
+                # Other guidance methods may return 1 batch (no guidance) or 3+ batches (e.g., PAG, APG).
+                guider_state = guider.prepare_inputs(guider_inputs, self._guider_input_fields)
+                # Step 4: Run the denoiser for each batch
+                # Each batch represents a different conditioning (conditional, unconditional, etc.).
+                # We run the model once per batch and store the noise prediction in each batch dict.
+                # After this loop, continuing the CFG example:
+                #   guider_state = [
+                #       {"encoder_hidden_states": prompt_embeds, "noise_pred": noise_pred_cond, "__guidance_identifier__": "pred_cond"},
+                #       {"encoder_hidden_states": negative_prompt_embeds, "noise_pred": noise_pred_uncond, "__guidance_identifier__": "pred_uncond"},
+                #   ]
+                for guider_state_batch in guider_state:
+                    guider.prepare_models(self.transformer)
+
+                    # Extract conditioning kwargs for this batch (e.g., encoder_hidden_states)
+                    cond_kwargs = guider_state_batch.as_dict()
+                    cond_kwargs = {k: v for k, v in cond_kwargs.items() if k in self._guider_input_fields}
+
+                    guider_state_batch_identifier = getattr(guider_state_batch, guider._identifier_key)
+                    with self.transformer.cache_context(guider_state_batch_identifier):
```

Comment:
A bit difficult for me to understand. 

In the Flux pipeline, we have "cond" context:
https://github.com/huggingface/diffusers/blob/f072c64bf22c6a259d38435b19a8ef3eb67c96a8/src/diffusers/pipelines/flux/pipeline_flux.py#L943

But here, we will have `"pred_cond"`. Is this expected?